frontend/src/lib/components/PayGateMini/payGateMiniLogic.tsx

```diff
@@ -0,0 +1,79 @@
+import { actions, connect, kea, key, path, props, selectors } from 'kea'
+import { billingLogic } from 'scenes/billing/billingLogic'
+import { preflightLogic } from 'scenes/PreflightCheck/preflightLogic'
+import { userLogic } from 'scenes/userLogic'
+
+import { AvailableFeature } from '~/types'
+
+import type { payGateMiniLogicType } from './payGateMiniLogicType'
+
+export interface PayGateMiniLogicProps {
+    featureKey: AvailableFeature
+    currentUsage?: number
+}
+
+export type GateVariantType = 'add-card' | 'contact-sales' | 'move-to-cloud' | null
+
+export const payGateMiniLogic = kea<payGateMiniLogicType>([
+    props({} as PayGateMiniLogicProps),
+    path(['lib', 'components', 'payGateMini', 'payGateMiniLogic']),
+    key((props) => props.featureKey),
+    connect(() => ({
+        values: [
+            billingLogic,
+            ['billing', 'billingLoading'],
+            userLogic,
+            ['user', 'hasAvailableFeature', 'availableFeature'],
+            preflightLogic,
+            ['isCloudOrDev'],
+        ],
+        actions: [],
+    })),
+    actions({
+        setGateVariant: (gateVariant: GateVariantType) => ({ gateVariant }),
+    }),
+    selectors(({ values, props }) => ({
+        product: [
+            (s) => [s.billing],
+            (billing) =>
+                billing?.products?.find((product) => product.features?.some((f) => f.key === props.featureKey)),
+        ],
+        featureInfo: [(s) => [s.product], (product) => product?.features.find((f) => f.key === props.featureKey)],
+        featureDetailsWithLimit: [
+            (s) => [s.user, (_, props) => props.featureKey],
+            (_user, featureKey) => {
+                return values.availableFeature(featureKey)
+            },
+        ],
+        minimumPlan: [
```

Comment:
Might be a bit verbose but I found the selector names confusing while reviewing so I'd prefer to lean on the side of verbosity.
```suggestion
        minimumPlanWithFeature: [
```